ledger/src/blockstore.rs

```diff
@@ -3922,20 +3922,36 @@ impl Blockstore {
         self.perf_samples_cf.put_bytes(index, &bytes)
     }
 
+    /// Returns the component vector for the slot starting with `shred_start_index`
+    pub fn get_slot_components(
+        &self,
+        slot: Slot,
+        shred_start_index: u64,
+    ) -> Result<Vec<BlockComponent>> {
+        self.get_slot_components_with_shred_info(slot, shred_start_index, false)
+            .map(|x| x.0)
+    }
+
     /// Returns the entry vector for the slot starting with `shred_start_index`
     pub fn get_slot_entries(&self, slot: Slot, shred_start_index: u64) -> Result<Vec<Entry>> {
```

Comment:
I tried to follow the call tree here to figure out how expensive this function might be.  

It seems like this function will call `get_slot_components_in_block()` which seems kinda expensive.  It looks up shreds from the DB; deshreds them; and then filters out the the `BlockComponent` and we then filter out the component with `Entry`s.  

Is it fair to say that this function can indeed be expensive?  Should add it to the documentation that that is the case?  Are there are other strategies possible to reduce the cost?